### PR TITLE
Make connection timeout user-configurable

### PR DESCRIPTION
**Pull request by haslinghuis**
Fixes: #2474 
Fixes: https://github.com/betaflight/betaflight-configurator/issues/416 

Configure connection timeout in Options tab for:

- bus initialization when user plugs in flight controller with auto-connect enabled 
- save and reboot

### CODE REVIEW

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:96]: Is this used only for autoconnect? Or when save and reboot is used too? If it's only for auto connect, I think we must use the word here. If it is for both, then the message is good.

**Review by McGiverGim — COMMENTED** [review]
Seems ok to me, only not too sure about the message.

File: src/js/port_handler.js
Comment (McGiverGim) [inline src/js/port_handler.js:183]: Maybe we need to move the `GUI.timeout_add` inside the `ConfigStorage.get()` callback to be sure that the `connectionTimeout` has been read?

File: locales/en/messages.json
Comment (haslinghuis) [reply locales/en/messages.json:96]: Yes good call, I think addressing save and reboot with this setting could be done too and could help tackle more connection issues.

**Review by McGiverGim — DISMISSED** [review]
To me it seems ok. Let's see how it works under user testing...

**Comment by mikeller** [comment]
This needs a rebase.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> PASS
 - PR is not in draft -> PASS
 - Assigned to a milestone -> FAIL
 - cooling off period lapsed (3/06/2021 23:42:47) -> PASS
 - commit count less or equal to three (1 commits) -> PASS
 - 'Don't merge' label NOT found -> PASS
 - 'Needs coordination with betaflight' label NOT found -> PASS
 - At least one 'RN:' label found -> FAIL
 - 'Tested' label found -> FAIL
 - assigned to an approver (None) -> FAIL
 - approver count at least three (0 approvers) -> FAIL

**Review by Asizon — APPROVED** [review]

File: src/js/port_handler.js
Comment (mikeller) [inline src/js/port_handler.js:178]: Looks like we are using the same setting here and in `serial_backend.js` to set two different timeouts, one of which is 15 times as long as the other by default - why are we setting it to the same, this does not seem to be making sense.

File: src/js/tabs/options.js
Comment (mikeller) [inline src/js/tabs/options.js:110]: Reading this from local storage in 3 different places makes this quite hard to maintain - this should probably be centralised.

**Review by mikeller — COMMENTED** [review]
Not sure this is the right approach - the required timeouts seem to be mostly dependent on the MCU / hardware configuration of the board used. So if making this settable is required, then the current implementation will force users using multiple different flight controllers to change this setting every time before they connect a different flight controller.
So I think better approaches for this will be:
- if there is one setting that works for all boards then just use this setting (or use it as the default, and expect nobody to ever change this option);
- if different settings are required for different boards, then use a more sophisticated approach, like re-trying on timeout.

File: src/js/tabs/options.js
Comment (haslinghuis) [reply src/js/tabs/options.js:110]: Yes this is lazy and reusing the same value on different occasions. But it works for some users as it's an option you would have to specify.

Comment (haslinghuis) [reply src/js/tabs/options.js:110]: Could add a `getConnectionTimeOut` function to centralize.

**Review by McGiverGim — APPROVED** [review]

File: src/js/tabs/options.js
Comment (haslinghuis) [reply src/js/tabs/options.js:110]: @mikeller Do you want me to make two separate values so this could be changed individual for both port_handler `auto-connect` and serial_backend `reinitializeConnection` and where should I centralize (put) the function `bool getconnectionTimeout(func)` to read the respective value?

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> FAIL
 - Assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - 'Don't merge' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by blckmn — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - Github identifies PR as mergeable -> FAIL
 - Assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - 'Don't merge' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - Github identifies PR as mergeable -> PASS
 - Assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - 'Don't merge' label NOT found -> PASS
 - At least one 'RN:' label found -> PASS
 - 'Tested' label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**
